src/factory.rs

```diff
@@ -9,16 +9,16 @@ pub fn factory<C: RuntimeName, I: ComInterface>() -> Result<I> {
 
     unsafe {
         // First attempt to get the activation factory via the OS.
-        let mut code = RoGetActivationFactory(name.get_abi(), &I::iid(), &mut factory);
+        let mut code = RoGetActivationFactory(name.get_abi(), &I::iid(), &mut factory)?;
 
         // If this fails because combase hasn't been loaded yet then load combase
         // automatically so that it "just works" for apartment-agnostic code.
         if code == ErrorCode::NOT_INITIALIZED {
             let mut _cookie = std::ptr::null_mut();
-            CoIncrementMTAUsage(&mut _cookie);
+            CoIncrementMTAUsage(&mut _cookie)?;
```

Comment:
This shouldn't fail either since it would force a Windows 7 project to create a COM apartment beforehand.

Reply:
Not completely sure I understand what needs to happen here and in on my phone ATM: this function isn't present on win7 at least, so to do that I would need to fall back to CoInitializeEx?

Reply:
Oh, nevermind about this one. For reg-free on Windows 7 the previous call to RoGetActivationFactory would not fail with NOT_INITIALIZED but with the GetProcAddress error code, so this should be fine.